modules/friendlytag.js

```diff
@@ -1711,18 +1761,32 @@ Twinkle.tag.callbacks = {
 			summaryText += ' {{[[:' + (tagName.indexOf(':') !== -1 ? tagName : 'Template:' + tagName + '|' + tagName) + ']]}}';
 		};
 
-		if (!tags.length) {
-			Morebits.status.warn('Info', 'No tags remaining to apply');
-		}
-
 		tags.sort();
 		$.each(tags, addTag);
+		if (rcatsToAdd.length && rcatsToRemove.length) {
+			summaryText += ' & ';
+		}
+		rcatsToRemove.forEach(function removeTag(tag) {
+			var tag_re = new RegExp('\\{\\{' + Morebits.pageNameRegex(tag) + '\\s*(\\|[^}]+)?\\}\\}\\n?');
+
+			if (tag_re.test(pageText)) {
+				pageText = pageText.replace(tag_re, '');
+			}
+			summaryText += 'Removed {{[[:' + (tag.indexOf(':') !== -1 ? tag : 'Template:' + tag + '|' + tag) + ']]}}';
```

Comment:
This should probably be sentence-like, as with article tags, with "Added ... and removed ..." rather than "Added... & Removed".  See line ~1310.  As it now stands, aside from that it adds "Removed" after every redirect, rather than listing them.